src/code/audio/code_8019AF00.c

```diff
@@ -1878,90 +1857,1124 @@ const u8 sIsOcarinaSongReserved[OCARINA_SONG_MAX] = {
+                // Checks to see if the song is reserved
+                if (sIsOcarinaSongReserved[i]) {
+                    // Loops through all possible starting indices
+                    for (j = 0; (j < 9 - gOcarinaSongButtons[i].numButtons); j++) {
+                        // Loops through the notes of the song
+                        for (k = 0; k < gOcarinaSongButtons[i].numButtons && k + j < 8 &&
+                                    gOcarinaSongButtons[i].buttonIndex[k] ==
+                                        gOcarinaSongButtons[OCARINA_SONG_SCARECROW_SPAWN].buttonIndex[k + j];
+                             k++) {
+                            ;
+                        }
+
+                        // This conditional is true if the recorded song contains a reserved song
+                        // recorded song is cancelled
+                        if (k == gOcarinaSongButtons[i].numButtons) {
+                            sRecordingState = OCARINA_RECORD_REJECTED;
+                            sOcarinaSongNotes[OCARINA_SONG_SCARECROW_SPAWN][1].volume = 0xFF;
+                            return;
+                        }
+                    }
+                }
+            }
+
+            // Counts how many times a note is repeated
+            i = 1;
+            while (i < 8) {
+                if (gOcarinaSongButtons[OCARINA_SONG_SCARECROW_SPAWN].buttonIndex[0] !=
+                    gOcarinaSongButtons[OCARINA_SONG_SCARECROW_SPAWN].buttonIndex[i]) {
+                    i = 9; // break
+                } else {
+                    i++;
+                }
+            }
+
+            // This condition is true if all 8 notes are the same pitch
+            if (i == 8) {
+                sRecordingState = OCARINA_RECORD_REJECTED;
+                sOcarinaSongNotes[OCARINA_SONG_SCARECROW_SPAWN][1].volume = 0xFF;
+                return;
+            }
+
+            // The scarecrow spawn song is accepted and copied from the buffer to the scarecrow spawn notes
+            for (i = 0; i < sRecordSongPos; i++) {
+                sOcarinaSongNotes[OCARINA_SONG_SCARECROW_SPAWN][i] = sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i];
+            }
+
+            sIsOcarinaInputEnabled = false;
+        } else {
+            sOcarinaSongNotes[OCARINA_SONG_SCARECROW_SPAWN][1].volume = 0xFF;
+        }
+    }
+
+    sRecordingState = OCARINA_RECORD_OFF;
+}
+
+/**
+ * recordingState = OCARINA_RECORD_OFF, end
+ * recordingState = OCARINA_RECORD_SCARECROW_LONG, start long scarecrows song
+ * recordingState = OCARINA_RECORD_SCARECROW_SPAWN, start spawn scarecrows song
+ */
+void AudioOcarina_SetRecordingState(u8 recordingState) {
+    if ((u32)recordingState == sRecordingState) {
+        return;
+    }
+
+    if (recordingState != OCARINA_RECORD_OFF) {
+        sOcarinaRecordTaskStart = sOcarinaUpdateTaskStart;
+        sRecordOcarinaPitch = OCARINA_PITCH_NONE;
+        sRecordOcarinaVolume = 0x57;
```

Comment:
Is there a logic to what goes in hex and what dec with this sort of thing?

Reply:
For volume, no and it's something that's also been bothering me for a while. Currently, hex is much more common but I've been contemplating changing it all to decimal. There's a whole rabbit-hole of nonsense with documenting volume though so I've procrastinated bringing it up, but it's on the to-do list to make consistent. For now though, should probably all be hex